fuzz: increase `txorphan` harness stability

marcofleon opened the pull request:
This moves `nNextSweep` from being a static variable in `LimitOrphans` to being a member of the `TxOrphanage` class. This improves the stability of the `txorphan` fuzz harness, as each orphanage (created every iteration) now has its own value for `nNextSweep`.

marcofleon committed:
increase txorphan harness stability

initialize variable

dergoegge commented on src/txorphanage.h:
```diff
@@ -107,7 +107,7 @@ class TxOrphanage {
 
     /** Timestamp for the next scheduled sweep of expired orphans */
-    NodeSeconds nNextSweep;
+    NodeSeconds n_next_sweep;
```
Should be `m_` not `n_`, see https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#coding-style-c

dergoegge reviewed (commented):
I suspect that the scripted diff is failing in CI but passing on your Mac because sed behaves differently on Mac vs Linux. Maybe [this](https://stackoverflow.com/questions/4247068/sed-command-with-i-option-failing-on-mac-but-works-on-linux) helps?

marcofleon commented:
Seems like CI still doesn't like it.

I tried this (no space after -i)
`sed -i'' 's/nNextSweep/m_next_sweep/g' $(git grep -l 'nNextSweep')`

and it didn't work on Mac. Adding the `-e` without `.bak` creates new `txorphanage` files with `-e` appended to the end of them. Something like `txorphanage.h-e`. Adding `.bak` prevented that but CI isn't accepting it.

dergoegge commented:
> I am not sure if a three line diff is worth a scripted diff

Just for context, I suggested it to marco as a learning exercise.

marcofleon committed:
scripted-diff: Replace nNextSweep with m_next_sweep

-BEGIN VERIFY SCRIPT-
sed -i 's/nNextSweep/m_next_sweep/g' $(git grep -l 'nNextSweep')
-END VERIFY SCRIPT-

fixing to match style

marcofleon commented on src/txorphanage.h:
```diff
@@ -104,6 +105,9 @@ class TxOrphanage {
     int EraseTxNoLock(const Wtxid& wtxid) EXCLUSIVE_LOCKS_REQUIRED(m_mutex);
+
+    /** Timestamp for the next scheduled sweep of expired orphans */
+    NodeSeconds m_next_sweep GUARDED_BY(m_mutex);
```
Got it, thanks. Should be fixed now.

marcofleon commented:
> You'll have to install and use gnu-sed on macos, or use Linux. If you don't want to test locally and want to know the syntax without testing, you can look at previous examples via `git log --grep='sed -i'`.

I installed gnu-sed and used that locally. CI looks good now.

dergoegge reviewed (approved):
Code review ACK 8defc182a31d828ad0f53ebf7e3be9e9cfc42d09

glozow reviewed (commented):
utACK 8defc182a31d828ad0f53ebf7e3be9e9cfc42d09, I can rebase on this pretty easily
